core/src/toga/style/pack.py

```diff
@@ -56,12 +60,12 @@
 DISPLAY_CHOICES = Choices(PACK, NONE)
 VISIBILITY_CHOICES = Choices(VISIBLE, HIDDEN)
 DIRECTION_CHOICES = Choices(ROW, COLUMN)
-ALIGNMENT_CHOICES = Choices(LEFT, RIGHT, TOP, BOTTOM, CENTER)
+ALIGN_ITEMS_CHOICES = Choices(LEFT, RIGHT, TOP, BOTTOM, CENTER)
```

Comment:
Can we implement this by simply making `left` and `top` aliases for `start`, and `right` and `bottom` aliases for `end`? This would be a change in the current documented behavior:

> If a value is provided, but the value isn’t honored, the alignment reverts to the default for the direction.

But any code that's setting invalid alignment options is broken anyway.

Reply:
There's an edge case if the direction is RTL as well (`left` and `start` aren't analogs if `direction=RTL`) - but I think I could live with both of those inconsistencies as long as using the old names raises an appropriate warning on first use.

Reply:
I haven't had time to get to it yet, but for maximum compatibility, what I envisioned was only storing whichever version was set last — `alignment` with directions, or `align_items` with `start`/`center`/`end`. Then if the same one is requested, return that, but if the other is requested, compute it based on the set value and the current row/column direction and RTL/LTR direction.

Reply:
I think it's possible to make `align_items` the single source of truth, while keeping full compatibility with the current `alignment` behavior:

* In the `alignment` setter, check the current `direction` and `text_direction`, and transform `top`, `bottom`, `left` or `right` to `start` or `end` according to the current behavior. In particular, this means that setting `alignment="bottom"` when `direction="column"` will result in `align_items="start"`, which is "the default for the direction".

* In the `alignment` getter, do the reverse transformation.

As discussed in #1194, there are some bugs in the layout algorithm. For example, the actual default alignment is sometimes `start` and sometimes what CSS calls `stretch`, and `center` doesn't always work. I'm working on fixing this in another branch which I'll stack on top of this one, so please don't make any structural changes to the layout algorithm beyond renaming the attributes.

Reply:
But what if existing code sets `alignment` and then changes `direction` or `text_direction`? This is why I think it's safer to just preserve the existing behavior for the old name. And if `align_items` is requested while `alignment` is currently an invalid value for the current `direction`/`text_direction`, then `align_items` can read as not having been set.

> As discussed in #1194, there are some bugs in the layout algorithm. For example, the actual default alignment is sometimes `start` and sometimes what CSS calls `stretch`, and `center` doesn't always work. I'm working on fixing this in another branch which I'll stack on top of this one, so please don't make any structural changes to the layout algorithm beyond renaming the attributes.

Thanks for the heads up! I was looking at potentially merging the nearly-identical row and column layout methods, either here or in a second PR, but will avoid mucking about with any of that for now.

Reply:
> for maximum compatibility, what I envisioned was only storing whichever version was set last. [...] Then if the same one is requested, return that, but if the other is requested, compute it

It would be more complex that way, but I guess it's the only way to achieve 100% backward compatibility. Either approach is OK with me.